### PR TITLE
Documents pack usage

### PR DESCRIPTION
**Pull request by ameyer-pivotal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,301 @@
...
+
+![](docs/create-builder-diagram.png)
+
+A builder is an image containing a collection of buildpacks that will be executed, in the order that they appear in
+`builder.toml`, against app source code. A buildpack's primary role is to inspect the source code, determine any
+dependencies that will be required to compile and/or run the app, and provide those dependencies as layers in the
+resulting image. This image's base will be the build image associated with a given stack.
+
+It's important to note that the buildpacks in a builder are not actually executed until
+[`create-builder`](#building-explained) is run.
```
Comment (sclevine) [inline README.md]: `pack build`?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,301 @@
...
+`pack build` enables app developers to create runnable app images from source code using buildpacks.
+
+```bash
+$ pack build <image-name>
 ```
-./pack build packs/myimage:mytag --path ./myapp
+
+### Example: Building Using the Default Builder Image
+
+In the following example, an app image is created from Node source code.
```
Comment (sclevine) [inline README.md]: "Node.js application source code"

File: README.md [hunk 3]
```diff
@@ -1,17 +1,301 @@
...
 ```
 
-The above will create images on your local daemon. If you wish to export images directly to a docker registry, use the `--publish` flag.
+In this case, the default [builder](#working-with-builders-using-create-builder) is used, and an appropriate buildpack
+is automatically selected from the builder based on the app source code. To understand more about what builders are and
+how to create or use them, see the
+[Working with Builders Using `create-builder`](#working-with-builders-using-create-builder) section.
 
+To publish the produced image to an image registry, include the `--publish` flag (this is most useful
+for private registries):
```
Comment (sclevine) [inline README.md]: This is actually more useful for public registries (where stack authors could maintain a run image).

File: README.md [hunk 4]
```diff
@@ -1,17 +1,301 @@
...
+
+```bash
+$ pack build private-registry.example.com/my-app:my-tag --publish
 ```
-./pack build packs/myimage:mytag --path ./myapp --publish
+
+### Example: Building Using a Specified Buildpack
+
+In the following example, an app image is created from Node source code, using a buildpack chosen by the user.
```
Comment (sclevine) [inline README.md]: "Node.js application source code"

File: README.md [hunk 5]
```diff
@@ -1,17 +1,301 @@
...
+
+The `--buildpack` parameter can be
+- a path to a directory
+- a path to a `.tgz` file
+- a URL to a `.tgz` file, or
+- the ID of a buildpack located in a builder
+
+### Building Explained
+
+![](docs/build-diagram.png)
```
Comment (sclevine) [inline README.md]: Add alt-text for accessibility.

File: README.md [hunk 6]
```diff
@@ -1,17 +1,301 @@
...
+```bash
+$ pack rebase my-app:my-tag
+```
+
+Like [`build`](#building-app-images-using-build), `rebase` has a `--publish` flag that can be
+used to publish the updated app image to a registry.
+
+### Rebasing Explained
+
+![](docs/rebase-diagram.png)
```
Comment (sclevine) [inline README.md]: alt-text for accessibility.

File: README.md (hunk 5 above)
Comment (sclevine) [reply README.md]: Also, might consider SVGs instead of raster images.

File: README.md (hunk 1 above)
Comment (ekcasey) [reply README.md]: the buildpacks in a builder aren't executed `pack build` is run

Comment (ameyer-pivotal) [reply README.md]: Oops! Copy/paste error. Thanks!

File: README.md (hunk 5 above)
Comment (ameyer-pivotal) [reply README.md]: @sclevine 
Re: alt-text - will do.
Re: SVGs, I've had trouble in the past putting SVGs in readmes on github, but I'll try it out

File: README.md [hunk 7]
```diff
@@ -1,17 +1,301 @@
...
+newer version of the app's base image exists (either locally or in a registry). If so, `rebase` updates the app image's
+layer metadata to reference the newer base image version.
+
+## Working with Builders Using `create-builder`
+
+`pack create-builder` enables buildpack authors and platform operators to bundle a collection of buildpacks into a
+single image for distribution and use with a specified stack.
+
+```bash
+$ pack create-builder --builder-config <path-to-builder-toml>
```
Comment (ssisil) [inline README.md]: pack create-builder my-builder-image:tag --builder-config <path-to-builder-toml>

Comment (ameyer-pivotal) [reply README.md]: @ssisil Good catch!

**Pull request merged**